test/validateNestedPoolState.test.ts

```diff
@@ -244,3 +260,66 @@ const deepPoolState: NestedPoolState = {
         },
     ],
 };
+
+const nestedOPoolState = {
```

Comment:
@brunoguerios I just created a quick test but naming should be adapted to better express the intention of the test. Could you improve it please?